### PR TITLE
Update gh action + document with roxygen2 + replace `T` -> `TRUE`, `F` by `FALSE` to satisfy R CMD CHECK

### PR DESCRIPTION
**Pull request by olivroy**
If it works, will close #326.

The failures appear to be related to an old version of the Matrix package that is cached. Maybe requiring a recent Matrix version in `DESCRIPTION` should solve the CI failures. I saw the same issues last month with the broom package.

On the first run, the Matrix version was 1.6-1.1 (released in September 2023).

I added a dependency on 1.6-2 (this can probably be reverted later if this causes issues)

### CODE REVIEW

**Comment by olivroy** [comment]
Actions seem fixed. However, the error seems weird. Is there somewhere in your workflow where you disallow using `T` instead of `TRUE`?  Because this is the error it encounters:

 Error in ifelse(absoluteDeviation == TRUE, "Residual spread [2*abs(res - 0.5)]",  : 
    T used instead of TRUE

**Comment by florianhartig** [comment]
It's actually good that it throws the error. The error is because of a CRAN policy that one can't use T in the help files. I believe this was not flagged on the old checks, usually this issue then came later when uploading to CRAN. 

I have pushed an update to main to fix this. 

The .RD files are created dynamically, so the change to the .RD doesn't do anything, but it also doesn't hurt!

Thanks a lot, this is very helpful!

**Comment by olivroy** [comment]
@florianhartig I updated the PR to change many `T` -> `TRUE` and `F` -> `FALSE`. CI should become green now!

File: DHARMa/DESCRIPTION [hunk 1]
```diff
@@ -21,7 +21,7 @@ Imports:
     graphics,
     utils,
     grDevices,
-    Matrix,
+    Matrix (>= 1.6-2),
```
Comment (olivroy) [inline DHARMa/DESCRIPTION:24]:
```suggestion
    Matrix,
```

**Comment by olivroy** [comment]
@florianhartig the current failure is due to 0.4.6.x being an incorrect version. I changed it to 0.4.6.9000 a version number commonly used by R package developers to signify a development version. https://r-pkgs.org/lifecycle.html#sec-lifecycle-version-number-tidyverse

**Comment by florianhartig** [comment]
Ok, this seems to be fine, will merge now, thanks for the help!

Fixes #326

**Comment by olivroy** [comment]
Thanks for merging! It got a bit bumpier than expected! Sorry about that. Cheers and let me know if  you have questions!

**Pull request merged**